iothub/device/src/Message.cs

```diff
@@ -28,21 +28,29 @@ public sealed class Message : IReadOnlyIndicator, IDisposable
         /// <summary>
         /// Default constructor with no body data
         /// </summary>
-        public Message()
+        /// <param name="setDefaultMessageId">A flag indicating if the MessageId should be set to a random GUID.</param>
+        public Message(bool setDefaultMessageId = false)
```

Comment:
@drwill-ms The messageId changes, as we discussed

Reply:
I envisioned it being a client options value, not one passed to each Message instance. Thoughts?

Reply:
The way I see it, client options are used to configure the client instance itself (proxy, model Id, http client etc,), whereas this is a configuration for a message. It isn't really related to a client per se.
Also, I might want some set of messages to have a library-set-guid for message Id, while others to have messageId as null. Taking this configuration at a message level will allow to user the option to configure it like that. (anything we set at client options level will apply to all messages).

Reply:
I see why you might see it that way.

The downsides are that:
- As a user, I have to pass true to each and every message. That is tedious.
- It makes the Message ctor signature messier.

I see it as, I tell the client I want it to behave a certain way when it does something, that could even extend to putting a message Id on each message, if not otherwise applied.